### PR TITLE
Configure nginx for client certificate authentication

### PR DESCRIPTION
**Pull request by Fadiabb**

### CODE REVIEW

File: docs/client-certificate-setup.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+If the certificate Authority and user certificates are already present the steps of creating the certificate authority and client certificates can be skipped.
+The following is an example of creating them.
+
+```bash
+cd /etc/ssl
+```
+### Create the Certificate Autority (CA)
+
+Firstly, generate the CA:      
+```openssl genrsa -des3 -out ca.key 4096```
```
Comment (bernhardreiter) [inline docs/client-certificate-setup.md:12]: Even when it is an example, we should only use ciphers that are recommened for new systems in  BSI's TR-02102-1 (https://www.bsi.bund.de/SharedDocs/Downloads/EN/BSI/Publications/TechGuidelines/TG02102/BSI-TR-02102-1.html)    -des3 is not , -aes256 is

File: docs/client-certificate-setup.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+```openssl genrsa -des3 -out ca.key 4096```          
+This asks to enter a passphrase.       
+Next, create the server-side certificate, that will be sent via the TLS server to the client.             
+```openssl req -new -x509 -days 365 -key ca.key -out ca.crt```       
+You will be asked to answer a few questions.
+
+### Create a client certificate
+
+Create the key like previously:              
+```openssl genresa -des3 -out userA.key 4906```
```
Comment (bernhardreiter) [inline docs/client-certificate-setup.md:21]: Typo genresa

**Comment by bernhardreiter** [comment]
We consider using ECC (if browsers allow) and gnutls for the scripted creation of key material.

**Comment by bernhardreiter** [comment]
Or even GnuPG for the creation of the certificates for something completely different. :) (But I'm not sure if this is documented, maybe doing a go application is even faster than doing this. ;) )

File: docs/client-certificate-setup.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+```bash
+cd /etc/ssl
+```
+### Create the Certificate Autority (CA)
+
+Firstly, generate the CA:      
+```openssl genrsa -aes256 -out ca.key 4096```          
+This asks to enter a passphrase.       
+Next, create the server-side certificate, that will be sent via the TLS server to the client.
```
Comment (tschmidtb51) [inline docs/client-certificate-setup.md:14]: Is that the CA cert or the server cert? If the latter, in most cases that will be an existing server certificate or from Let's Encrypt. Please take that into consideration and clarify/differentiate which steps are needed when using an existing server cert.

File: docs/client-certificate-setup.md (hunk 1 above)
Comment (bernhardreiter) [reply docs/client-certificate-setup.md]: (Infos now in other document)

**Comment by bernhardreiter** [comment]
This branch now also implements the checking for tls certificates in the `csaf_provider` cgi binary.

What is missing here or elsewhere is to be more clear in the configuration about where the document root for nginx is and should be.

File: cmd/csaf_provider/controller.go [hunk 4]
```diff
@@ -69,15 +70,24 @@ func (c *controller) bind(pim *pathInfoMux) {
...
-	}
 	return func(rw http.ResponseWriter, r *http.Request) {
-		hash := r.Header.Get("X-CSAF-PROVIDER-AUTH")
-		if !c.cfg.checkPassword(hash) {
+
+		verify := os.Getenv("SSL_CLIENT_VERIFY")
+		log.Printf("SSL_CLIENT_VERIFY: %s\n", verify)
+
+		switch {
+		case verify == "SUCCESS" && os.Getenv("SSL_CLIENT_I_DN") == c.cfg.Issuer:
```
Comment (bernhardreiter) [inline cmd/csaf_provider/controller.go:79]: Where is the documentation for this config options, especially the format it accepts.

Comment (bernhardreiter) [reply cmd/csaf_provider/controller.go]: @Fadiabb please add that documentation.

**Pull request merged**